presto-benchmark/src/main/java/com/facebook/presto/benchmark/PageProcessorBenchmark.java

```diff
@@ -0,0 +1,174 @@
+import com.facebook.presto.spi.type.Type;
+import com.facebook.presto.sql.gen.ExpressionCompiler;
+import com.facebook.presto.sql.relational.ConstantExpression;
+import com.facebook.presto.sql.relational.RowExpression;
+import com.facebook.presto.testing.TestingSession;
+import com.google.common.collect.ImmutableList;
+import org.jetbrains.annotations.NotNull;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Param;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.Warmup;
+import org.openjdk.jmh.infra.Blackhole;
+import org.openjdk.jmh.runner.Runner;
+import org.openjdk.jmh.runner.RunnerException;
+import org.openjdk.jmh.runner.options.Options;
+import org.openjdk.jmh.runner.options.OptionsBuilder;
+import org.openjdk.jmh.runner.options.VerboseMode;
+
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import static com.facebook.presto.metadata.FunctionKind.SCALAR;
+import static com.facebook.presto.metadata.MetadataManager.createTestMetadataManager;
+import static com.facebook.presto.spi.type.BigintType.BIGINT;
+import static com.facebook.presto.sql.relational.Expressions.call;
+import static com.facebook.presto.sql.relational.Expressions.constant;
+import static com.facebook.presto.sql.relational.Expressions.field;
+import static java.util.stream.Collectors.toList;
+
+@State(Scope.Thread)
+@OutputTimeUnit(TimeUnit.MICROSECONDS)
+@Fork(1)
+@Warmup(iterations = 3)
+@Measurement(iterations = 5)
+@BenchmarkMode(Mode.AverageTime)
+public class PageProcessorBenchmark
+{
+    private static final Metadata METADATA = createTestMetadataManager();
+    private static final Session TEST_SESSION = TestingSession.testSessionBuilder().build();
+
+    private PageProcessor processor;
+    private List<Page> inputPages;
+
+    private List<? extends Type> types;
+
+    @Param({ "10", "100", "1000", "2000" })
+    int pageCount;
+
+    @Param({ "1000", "2000", "4000" })
+    int positions;
+
+    @Setup
+    public void setup()
+    {
+        List<RowExpression> projections = getProjections();
+        types = projections.stream().map(RowExpression::getType).collect(toList());
+
+        inputPages = createPages(pageCount);
+        processor = new ExpressionCompiler(createTestMetadataManager()).compilePageProcessor(getFilter(), projections);
+    }
+
+    @Benchmark
+    public void rowOriented(Blackhole b)
+    {
+        for (int i = 0; i < pageCount; i++) {
+            PageBuilder pageBuilder = createPageBuilder();
+            b.consume(processor.process(null, inputPages.get(i), 0, inputPages.get(i).getPositionCount(), pageBuilder));
```

Comment:
Also, it's generally not a good idea to "loop" in the benchmark code, but rely on jmh to do that for you. Otherwise, you may see effects of loop unrolling and so on.